interfaces/builtin/system_files.go

```diff
@@ -0,0 +1,71 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2018 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"fmt"
+	"strings"
+)
+
+const systemFilesSummary = `allows access to system files or directories`
+
+const systemFilesBaseDeclarationSlots = `
+  system-files:
+    allow-installation: false
+    deny-connection: true
+    deny-auto-connection: true
+`
+
+const systemFilesConnectedPlugAppArmor = `
+# Description: Can access specific system files or directories.
+# This is restricted because it gives file access to arbitrary locations.
+`
+
+type systemFilesInterface struct {
+	commonFilesInterface
+}
+
+func validateSinglePathSystem(np string) error {
+	if !strings.HasPrefix(np, "/") {
+		return fmt.Errorf(`%q must start with "/"`, np)
+	}
+	if strings.Contains(np, "$HOME") {
```

Comment:
I think here we should check $HOME is not a prefix, right?

Reply:
For system-files, we don't want $HOME at all, so the simple Contains check here is correct.